src/core/random.c

```diff
@@ -8,18 +8,29 @@ FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL );
 
 #include <stdlib.h>
 #include <ipxe/timer.h>
+#include <ipxe/profile.h>
 
 static int32_t rnd_seed = 0;
 
 /**
  * Seed the pseudo-random number generator
  *
  * @v seed		Seed value
+ *
+ * Will use the following entropy sources if @c seed is 0:
+ *  - system time ticks
+ *  - cpu profiling timestamp
+ *  - address of stack variable
+ *  - fair dice roll (ensures nonzero seed)
  */
 void srandom ( unsigned int seed ) {
-	rnd_seed = seed;
-	if ( ! rnd_seed )
-		rnd_seed = 4; /* Chosen by fair dice roll */
+	if ( ! ( rnd_seed = seed ) ) {
+		rnd_seed = ( currticks()
+			^ profile_timestamp()
+			^ ( size_t ) &seed
+		) | 4;
```

Comment:
Please keep the `/* Chosen by fair dice roll */`

Or add the `alt_text` from https://xkcd.com/221/
> "RFC 1149.5 specifies 4 as the standard IEEE-vetted random number."

Reply:
Sure, I moved it into the doxygen comment instead, but keeping an additional inline copy is aight with me :+1: